server/text_generation_server/utils/layers.py

```diff
@@ -467,46 +467,159 @@ def forward(self, x):
         return x
 
 
-class SpeculativeHead(nn.Module):
+class MedusaHeadV1(nn.Module):
```

Comment:
Maybe it's time we put them into their own file (both in the same file is ok I think)?